### PR TITLE
RDM Responder cleanup & bugfixes.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: firmware/src/rdm.h [hunk 1]
```diff
@@ -59,6 +59,17 @@ static const uint16_t SUBDEVICE_MAX = 0x0200;
 static const uint8_t SUB_START_CODE = 0x01;
 
 /**
+<<<<<<< HEAD
```
Comment (peternewman) [inline firmware/src/rdm.h]: Fix the merge!

File: firmware/src/rdm_responder.c [hunk 2]
```diff
@@ -98,7 +100,7 @@ static void Checksum(const IOVec* iov, unsigned int iov_count,
  */
 static void SendDUBResponseIfRequired(const uint8_t *param_data,
                                       unsigned int param_data_length) {
-  if (param_data_length != 2 * UID_LENGTH) {
+  if (param_data_length != 2 * UID_LENGTH || g_rdm_responder.is_muted) {
```
Comment (peternewman) [inline firmware/src/rdm_responder.c]: Perhaps add this first to avoid the calculation?

File: firmware/src/rdm_responder.c [hunk 3]
```diff
@@ -227,8 +225,9 @@ static void Mute(const RDMHeader *header,
   }
   g_rdm_responder.is_muted = true;
 
+  memset(g_rdm_responder.param_data, 0, 8);
```
Comment (peternewman) [inline firmware/src/rdm_responder.c]: Where does the 8 come from, can you sizeof?

File: tests/tests/RDMResponderTest.cpp [hunk 4]
```diff
@@ -61,6 +61,50 @@ void SendResponse(bool include_break,
...
+
+TEST_P(ChecksumTest, checksumFails) {
+  EXPECT_FALSE(RDMResponder_VerifyChecksum(sample_message, GetParam()));
+}
+
+INSTANTIATE_TEST_CASE_P(
+    sizeTooSmall,
+    ChecksumTest,
+    ::testing::Range(
+      0u,
```
Comment (peternewman) [inline tests/tests/RDMResponderTest.cpp]: Alignment

File: tests/tests/RDMResponderTest.cpp [hunk 5]
```diff
@@ -61,6 +61,50 @@ void SendResponse(bool include_break,
...
+      static_cast<unsigned int>(arraysize(ChecksumTest::sample_message) - 1)));
```
Comment (peternewman) [inline tests/tests/RDMResponderTest.cpp]: Again

**Comment by peternewman** [comment]
LGTM aside from comments

File: firmware/src/rdm.h (hunk 1 above)
Comment (nomis52) [inline firmware/src/rdm.h]: Done

File: firmware/src/rdm_responder.c (hunk 2 above)
Comment (nomis52) [inline firmware/src/rdm_responder.c]: Done

File: firmware/src/rdm_responder.c (hunk 3 above)
Comment (nomis52) [inline firmware/src/rdm_responder.c]: Done

File: tests/tests/RDMResponderTest.cpp (hunk 4 above)
Comment (nomis52) [inline tests/tests/RDMResponderTest.cpp]: Done

File: firmware/src/rdm_responder.c (hunk 3 above)
Comment (peternewman) [inline firmware/src/rdm_responder.c]: Didn't realise we never actually use the fields...

**Comment by peternewman** [comment]
LGTM

**Pull request merged**